### PR TITLE
Feature add bounty to phase

### PR DESCRIPTION
**Pull request by Vayras**
## Describe your changes
added a property phase_uuid and phase_priority to bounties in order to relate them to phases within features.

## Issue ticket number and link
#1662

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code

### CODE REVIEW

File: handlers/bounty.go [hunk 1]
```diff
@@ -229,6 +229,18 @@ func (h *bountyHandler) CreateOrEditBounty(w http.ResponseWriter, r *http.Reques
 		bounty.Tribe = "None"
 	}
 
+	// if bounty.PhaseUuid == nil {
+	// 	w.WriteHeader(http.StatusBadRequest)
+	// 	json.NewEncoder(w).Encode("PhaseUuid is a required field")
```
Comment (elraphty) [inline handlers/bounty.go:234]: @Vayras Please remove this commented lines of code.

Comment (Vayras) [reply handlers/bounty.go:234]: @elraphty sure thing

**Comment by elraphty** [comment]
@Vayras You didn't assert for anything in the cypress test, and you were supposed to work on that https://github.com/stakwork/sphinx-tribes/blob/7e4d6d3ffe1dc37300707c4e64b18940063ae9b4/cypress/e2e/07_add_bounty_to_phase.cy.ts

**Comment by Vayras** [comment]
> @Vayras You didn't assert for anything in the cypress test, and you were supposed to work on that https://github.com/stakwork/sphinx-tribes/blob/7e4d6d3ffe1dc37300707c4e64b18940063ae9b4/cypress/e2e/07_add_bounty_to_phase.cy.ts

@elraphty looks like the assertions were not updated in the commits, let me push another commit

**Comment by elraphty** [comment]
> > @Vayras You didn't assert for anything in the cypress test, and you were supposed to work on that https://github.com/stakwork/sphinx-tribes/blob/7e4d6d3ffe1dc37300707c4e64b18940063ae9b4/cypress/e2e/07_add_bounty_to_phase.cy.ts
> 
> @elraphty looks like the assertions were not updated in the commits, let me push another commit

Okay, thanks.

**Comment by elraphty** [comment]
@Vayras please make the PR to this branch `feature-add-bounty-to-phase)` and not `master`

**Comment by elraphty** [comment]
@Vayras please rebase to `[feature-add-bounty-to-phase](https://github.com/stakwork/sphinx-tribes/tree/feature-add-bounty-to-phase)` to get the accurate test conditions

**Comment by Vayras** [comment]
> @Vayras please rebase to `[feature-add-bounty-to-phase](https://github.com/stakwork/sphinx-tribes/tree/feature-add-bounty-to-phase)` to get the accurate test conditions

@elraphty sure will do , currently away from desk will get this done as soon as possible

**Comment by Vayras** [comment]
@elraphty @fvalentiner 4 tests passed working on the other two
Screenshot 2024-05-23 at 11 19 54 PM

**Comment by elraphty** [comment]
@Vayras  this test is failing

```sh

//This test initially does not pass! It asserts that the endpoint should not receive a phase_uuid that doesn't exist
describe('Create Bounties with wrong phase_uuid', () => {
    it('passes', () => {
        cy.upsertlogin(User).then(value => {
            cy.request({
                method: 'POST',
                url: `${HostName}/gobounties/`,
                headers: { 'x-jwt': `${value}` },
                body: {...Bounties[0], phase_uuid: 'cp68lagn1e462l489mu0'}, //phase_uuid does not exist
                failOnStatusCode: false
            }).then((resp) => {
                expect(resp.status).to.eq(400);
            })
        })
    })
});
```

**Comment by Vayras** [comment]
> @Vayras  this test is failing
> 
> ```sh
> 
> //This test initially does not pass! It asserts that the endpoint should not receive a phase_uuid that doesn't exist
> describe('Create Bounties with wrong phase_uuid', () => {
>     it('passes', () => {
>         cy.upsertlogin(User).then(value => {
>             cy.request({
>                 method: 'POST',
>                 url: `${HostName}/gobounties/`,
>                 headers: { 'x-jwt': `${value}` },
>                 body: {...Bounties[0], phase_uuid: 'cp68lagn1e462l489mu0'}, //phase_uuid does not exist
>                 failOnStatusCode: false
>             }).then((resp) => {
>                 expect(resp.status).to.eq(400);
>             })
>         })
>     })
> });
> ```

@elraphty working on this

**Comment by elraphty** [comment]
> > @Vayras  this test is failing
> > ```shell
> > //This test initially does not pass! It asserts that the endpoint should not receive a phase_uuid that doesn't exist
> > describe('Create Bounties with wrong phase_uuid', () => {
> >     it('passes', () => {
> >         cy.upsertlogin(User).then(value => {
> >             cy.request({
> >                 method: 'POST',
> >                 url: `${HostName}/gobounties/`,
> >                 headers: { 'x-jwt': `${value}` },
> >                 body: {...Bounties[0], phase_uuid: 'cp68lagn1e462l489mu0'}, //phase_uuid does not exist
> >                 failOnStatusCode: false
> >             }).then((resp) => {
> >                 expect(resp.status).to.eq(400);
> >             })
> >         })
> >     })
> > });
> > ```
> 
> @elraphty working on this

Okay, thanks.

**Comment by Vayras** [comment]
@elraphty fixed all the tests
Screenshot 2024-05-24 at 2 41 52 PM

**Comment by elraphty** [comment]
@Vayras the Go test, and Cypress Test are failing

**Comment by Vayras** [comment]
> @Vayras the Go test, and Cypress Test are failing

@elraphty the Cypress test has passed locally, have to check the problem here

**Pull request merged**